lightningd/channel/channel.c

```diff
@@ -81,6 +84,20 @@ static void queue_pkt(struct peer *peer, const u8 *msg)
 	io_wake(peer);
 }
 
+static struct io_plan *gossip_client_recv(struct io_conn *conn,
+					  struct daemon_conn *dc)
+{
+	u8 *msg = dc->msg_in;
+	struct peer *peer = container_of(dc, struct peer, gossip_client);
+	u16 type = fromwire_peektype(msg);
+
+	if (type == WIRE_CHANNEL_ANNOUNCEMENT || type == WIRE_CHANNEL_UPDATE ||
+	    type == WIRE_NODE_ANNOUNCEMENT)
+		queue_pkt(peer, msg);
+
+	return daemon_conn_read_next(conn, dc);
```

Comment:
Shouldn't we fail if the gossip daemon sends an unknown packet?  (Well, we should also check them for well-formedness, I suppose).